client/lib/feed-stream-store/index.js

```diff
@@ -45,6 +45,11 @@ function mixedKeyMaker( post ) {
 	return siteKeyMaker( post );
 }
 
+const recommendedKeyMaker = post => ( {
+	...siteKeyMaker( post ),
+	isRecommendation: true }
```

Comment:
Can you explain your logic here regarding how it is cluttering the API?  From what I can tell, the original intent of the `postKey` was to be an object that could directly key into our fetch post APIs ( and to 1:1 uniquely identify a single post).  `gaps` only sort-of violate the constraint whereas the `is*` fully violates it.  I do agree that I am further breaking it by adding even more ui logic to the `postKey`.

In my summary at the top I allude to this here:
> feed-stream-store now adds is_recommendation: true to any post that comes from a recommendation api. I think ideally in the future we can fully separate out the notion of where the data comes from, how to display it, and how to report metrics on it.


It is on my todo list to figure out how to separate out our ui rendering logic from our data fetching logic, but I actually think that this consolidation is a step in the right direction and even if i'm breaking the postKey concept further.  I'm hoping we can create a redux store containing logic re. how to render each individual card (probably will need to be synced up with streams) and then the post-lifecycle can `connect` to that.